x-pack/plugins/ml/server/routes/trained_models.ts

```diff
@@ -100,20 +104,50 @@ export function trainedModelsRoutes({ router, routeGuard }: RouteInitialization)
                   ...Object.values(modelDeploymentsMap).flat(),
                 ])
               );
+              const modelsClient = modelsProvider(client);
 
-              const pipelinesResponse = await modelsProvider(client).getModelsPipelines(
-                modelIdsAndAliases
+              const modelsPipelinesAndIndices = await Promise.all(
+                modelIdsAndAliases.map(async (modelIdAndAlias) => {
+                  return {
+                    modelIdAndAlias,
+                    result: await modelsClient.getModelsPipelinesAndIndicesMap(modelIdAndAlias),
+                  };
+                })
               );
+
               for (const model of result) {
-                model.pipelines = {
-                  ...(pipelinesResponse.get(model.model_id) ?? {}),
-                  ...(model.metadata?.model_aliases ?? []).reduce((acc, alias) => {
-                    return Object.assign(acc, pipelinesResponse.get(alias) ?? {});
-                  }, {}),
-                  ...(modelDeploymentsMap[model.model_id] ?? []).reduce((acc, deploymentId) => {
-                    return Object.assign(acc, pipelinesResponse.get(deploymentId) ?? {});
-                  }, {}),
-                };
+                const modelAliases = model.metadata?.model_aliases ?? [];
+                const modelMap = modelsPipelinesAndIndices.find(
+                  (d) => d.modelIdAndAlias === model.model_id
+                )?.result;
+
+                const allRelatedModels = modelsPipelinesAndIndices
+                  .filter(
+                    (m) =>
+                      [
+                        model.model_id,
+                        ...modelAliases,
+                        ...(modelDeploymentsMap[model.model_id] ?? []),
+                      ].findIndex((alias) => alias === m.modelIdAndAlias) > -1
+                  )
+                  .map((r) => r?.result)
+                  .filter(isDefined);
+                const ingestPipelinesFromModelAliases = allRelatedModels
+                  .map((r) => r?.ingestPipelines)
+                  .filter(isDefined) as Array<Map<string, Record<string, PipelineDefinition>>>;
+
+                model.pipelines = ingestPipelinesFromModelAliases.reduce<
+                  Record<string, PipelineDefinition>
+                >((allPipelines, pipelineDefinitions) => {
+                  for (const [, value] of pipelineDefinitions?.entries()) {
+                    allPipelines = { ...allPipelines, ...(value ?? {}) };
```

Comment:
we should avoid spread operator in reduce.

Reply:
Updated here [`edc8a8c` (#162853)](https://github.com/elastic/kibana/pull/162853/commits/edc8a8c3eff3c8ec0c2623c7ca08be529556dc36)